src/Uno.UI/UI/Xaml/Shapes/Shape.iOSmacOS.cs

```diff
@@ -0,0 +1,902 @@
+			base.BackgroundColor = SolidColorBrushHelper.Transparent.Color;
+#elif __MACOS__
+			base.WantsLayer = true;
+			base.Layer.BackgroundColor = SolidColorBrushHelper.Transparent.Color;
+#endif
+		}
+
+		protected override void OnBackgroundChanged(DependencyPropertyChangedEventArgs e)
+		{
+			// Don't call base, we need to keep UIView.BackgroundColor set to transparent
+		}
+
+		#region Measure / Arrange should be shared using Geometry instead of CGPath
+		private protected Size MeasureRelativeShape(Size availableSize)
+		{
+			var stretch = Stretch;
+			var userMinSize = new Size(MinWidth, MinHeight);
+			var userSize = new Size(Width, Height);
+
+			var size = userSize;
+
+			// If no user size defined on a given axis, we try to stretch along this axis
+			if (IsNaN(size.Width))
+			{
+				size.Width = stretch == Stretch.UniformToFill || HorizontalAlignment == HorizontalAlignment.Stretch
+					? availableSize.Width
+					: 0;
+			}
+			if (IsNaN(size.Height))
+			{
+				size.Height = stretch == Stretch.UniformToFill || VerticalAlignment == VerticalAlignment.Stretch
+					? availableSize.Height
+					: 0;
+			}
+
+			// In case userSize was not defined, we still have to apply the min size
+			size = size
+				.AtLeast(userMinSize)
+				.NumberOrDefault(userMinSize);
+
+			if (IsInfinity(size.Width) || IsInfinity(size.Height))
+			{
+				// The size is invalid (the userSize was not defined and we were not able to stretch), just hide the shape.
+				// Note: This will be overriden by the Layouter that will enforce the MinSize
+				return default;
+			}
+			else
+			{
+				return size;
+			}
+		}
+		private protected (Size shapeSize, Rect renderingArea) ArrangeRelativeShape(Size finalSize)
+		{
+			var horizontal = HorizontalAlignment;
+			var vertical = VerticalAlignment;
+			var stretch = Stretch;
+			var userMinSize = new Size(MinWidth, MinHeight);
+			var userMaxSize = new Size(MaxWidth, MaxHeight);
+			var userSize = new Size(Width, Height);
+
+			var size = userSize;
+
+			// Like for the measure, if no user size defined on a given axis, we try to stretch along this axis
+			if (IsNaN(size.Width))
+			{
+				size.Width = stretch == Stretch.UniformToFill || HorizontalAlignment == HorizontalAlignment.Stretch
+					? finalSize.Width
+					: 0;
+			}
+			if (IsNaN(size.Height))
+			{
+				size.Height = stretch == Stretch.UniformToFill || VerticalAlignment == VerticalAlignment.Stretch
+					? finalSize.Height
+					: 0;
+			}
+
+			// Like for the measure, in case userSize was not defined, we still have to apply the min size
+			size = size
+				.AtLeast(userMinSize)
+				.NumberOrDefault(userMinSize);
+
+			// The area that will be used to render the rectangle/ellipse as path
+			var renderingArea = new Rect(new Point(), size);
+
+			// Apply the stretch mode, as it might change the "shape" of a "relative shape"
+			switch (stretch)
+			{
+				case Stretch.None:
+					renderingArea.Height = renderingArea.Width = 0;
+					break;
+
+				default:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Move this 'default:' case to the beginning or end of this 'switch' statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5633443)